recipe/meta.yaml

```diff
@@ -92,7 +92,12 @@ outputs:
         - echo 'ok'  # [not osx]
         # make CONDA_BUILD_SYSROOT a 'used variable'
         # so that conda-build knows to pass it to the test env
-        - echo "{{ CONDA_BUILD_SYSROOT }}"  # [osx]
+        {% if CONDA_BUILD_SYSROOT is defined %}      # [osx]
+        - echo "{{ CONDA_BUILD_SYSROOT }}"           # [osx]
+        {% else %}                                   # [osx]
+        - echo "CONDA_BUILD_SYSROOT is undefined"    # [osx]
+        - exit 1                                     # [osx]
+        {% endif %}                                  # [osx]
```

Comment:
Starting with conda-build 24.7.0 the Jinja parsing has become more strict. This guards the usage of `CONDA_BUILD_SYSROOT` to protect against issues with it

That said, am curious about the reason this is here in the first place. Appears this was added some time ago ( https://github.com/conda-forge/openmpi-feedstock/pull/35 ). Is this workaround still needed?

Reply:
Thanks for digging into this, John! IIUC @minrk added them in #35 which predated my time as maintainer. I am supportive in removing them because it doesn't seem natural to me that we need to touch `CONDA_BUILD_SYSROOT` in a recipe, but @minrk would know the best.

Reply:
Testing this out in commit: https://github.com/conda-forge/openmpi-feedstock/pull/177/commits/aaa2fbd1102219b933c5a25da40c57317a8dd6bc

Reply:
Wonderful!